cmd/kind/kind.go

```diff
@@ -119,22 +119,49 @@ func Run() error {
 	return NewCommand().Execute()
 }
 
+// Main wraps Run and sets the log formatter
+func Main() {
+	if err := Run(); err != nil {
+		logError(err)
+		os.Exit(1)
+	}
+}
+
 // logError logs the error and the root stacktrace if there is one
 func logError(err error) {
 	globals.GetLogger().Errorf("ERROR: %v", err)
-	// display stack trace if the error has one
+	// display the stack trace if any
+	if trace := stackTrace(err); trace != nil {
+		globals.GetLogger().Errorf("%+v", trace)
+	}
+}
+
+// stackTrace returns the deepest StackTrace is a Cause chain
+// https://github.com/pkg/errors/issues/173
+func stackTrace(err error) errors.StackTrace {
+	// github.com/pkg/errors errors type interfaces
+	type causer interface {
+		Cause() error
+	}
 	type stackTracer interface {
 		StackTrace() errors.StackTrace
 	}
-	if tracer, ok := errors.Cause(err).(stackTracer); ok {
```

Comment:
Cause always returns the deepest error in the chain, which will frequently be some error that was wrapped and does NOT include a stack trace. We now find the deepest error in the wrap chain with a trace and display that trace.

We still need better management of when traces are displayed, but we can handle that better in future revisions...